python/protobuf.c

```diff
@@ -254,8 +255,16 @@ static const char *PyUpb_GetClassName(PyType_Spec *spec) {
 PyTypeObject *PyUpb_AddClass(PyObject *m, PyType_Spec *spec) {
   PyObject *type = PyType_FromSpec(spec);
   const char *name = PyUpb_GetClassName(spec);
-  return type && PyModule_AddObject(m, name, type) == 0 ? (PyTypeObject *)type
-                                                        : NULL;
+  int result = PyModule_AddObject(m, name, type);
+  return type && result == 0 ? (PyTypeObject*)type : NULL;
+}
+
+PyTypeObject* PyUpb_AddClassWithBases(PyObject* m, PyType_Spec* spec,
+                                      PyObject* bases) {
+  PyObject* type = PyType_FromSpecWithBases(spec, bases);
+  const char* name = PyUpb_GetClassName(spec);
+  int result = PyModule_AddObject(m, name, type);
+  return type && result == 0 ? (PyTypeObject*)type : NULL;
```

Comment:
I would rather push this cast into the variable declaration and have a simple ternary

Reply:
If the declaration is `PyObject` then other uses of `type` require a cast. I realized I needed to rework the error handling on this a bit, take a look at the new code and see what you think.